nixos/modules/services/networking/opkssh/opkssh.nix

```diff
@@ -0,0 +1,171 @@
+      description = "System group for opkssh";
+    };
+
+    providers = mkOption {
+      type = types.attrsOf (
+        types.submodule {
+          options = {
+            issuer = mkOption {
+              type = types.str;
+              description = "Issuer URI";
+              example = "https://accounts.google.com";
+            };
+
+            clientId = mkOption {
+              type = types.str;
+              description = "OAuth client ID";
+            };
+
+            lifetime = mkOption {
+              type = types.str;
+              default = "24h";
+              description = "Token lifetime";
+            };
+          };
+        }
+      );
+      default = {
+        google = {
+          issuer = "https://accounts.google.com";
+          clientId = "206584157355-7cbe4s640tvm7naoludob4ut1emii7sf.apps.googleusercontent.com";
+          lifetime = "24h";
+        };
+        microsoft = {
+          issuer = "https://login.microsoftonline.com/9188040d-6c67-4c5b-b112-36a304b66dad/v2.0";
+          clientId = "096ce0a3-5e72-4da8-9c86-12924b294a01";
+          lifetime = "24h";
+        };
+      };
+      description = "OpenID Connect providers configuration";
+      example = literalExpression ''
+        {
+          google = {
+            issuer = "https://accounts.google.com";
+            clientId = "206584157355-7cbe4s640tvm7naoludob4ut1emii7sf.apps.googleusercontent.com";
+            lifetime = "24h";
+          };
+          github = {
+            issuer = "https://token.actions.githubusercontent.com";
+            clientId = "github";
+            lifetime = "oidc";
+          };
+        }
+      '';
+    };
+
+    authorizations = mkOption {
+      type = types.listOf (
+        types.submodule {
+          options = {
+            user = mkOption {
+              type = types.str;
+              description = "Linux user to authorize";
+            };
+
+            principal = mkOption {
+              type = types.str;
+              description = "Principal identifier (email, repo, etc.)";
+            };
+
+            issuer = mkOption {
+              type = types.str;
+              description = "Issuer URI";
+            };
+          };
+        }
+      );
+      default = [ ];
+      description = "User authorization mappings";
+      example = literalExpression ''
+        [
+          {
+            user = "github";
+            principal = "example@gmail.com";
+            issuer = "https://accounts.google.com";
+          }
+          {
+            user = "github";
+            principal = "repo:NixOs/nixpkgs:environment:production";
+            issuer = "https://token.actions.githubusercontent.com";
+          }
+        ]
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.groups.${cfg.group} = { };
+    users.users.${cfg.user} = {
+      isSystemUser = true;
+      description = "OpenPubkey OpenID Connect SSH User";
+      group = cfg.group;
+    };
```

Comment:
I'm wondering if this will need a pinned UID:GID

See: `nixos/modules/misc/ids.nix`

Reply:
In the linked document it states: "If DynamicUser is problematic consider making a `isSystemUser=true` user with the uid and gid unset and let NixOS pick dynamic persistent ids on activation."

So I guess what we are doing here is fine? and we do not need to add a static ID to the list..

Reply:
Yeah this is mostly about least priviledge see [opkssh install script](https://github.com/openpubkey/opkssh/blob/a6119cbc2a788629f44af807d2a1213a84ae9da7/scripts/install-linux.sh#L346-L363), there is no well-known number required.